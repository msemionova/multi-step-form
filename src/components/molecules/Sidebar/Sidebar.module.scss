.sidebar {
  padding: 3.8rem;
  background-size: cover;
  background: no-repeat center/100% url("~/public/assets/images/bg-sidebar-desktop.svg");
  width: 27.4rem;
  min-width: 27.4rem;
  min-height: 57rem;
  flex: 0 0 auto;
}
