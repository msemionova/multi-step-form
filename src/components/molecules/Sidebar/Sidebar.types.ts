export type SidebarProps = {
  data: {
    [key: string]: unknown;
  }[];
};
