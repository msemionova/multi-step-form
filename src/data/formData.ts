import { FormDataProps } from './formData.types';

const formData: FormDataProps = [
  {
    name: 'Your info',
    title: 'Personal info',
    subtitle: 'Please provide your name, email address, and phone number.',
    fieldGroups: [
      {
        type: 'inputs',
        fields: [
          {
            id: 'username',
            label: 'Name',
            placeholder: 'e.g. John Doe',
            minlength: 2,
            maxlength: 50,
            required: true
          },
          {
            id: 'email',
            label: 'Email Address',
            type: 'email',
            placeholder: 'e.g. johndoe@email.com',
            required: true
          },
          {
            id: 'phone',
            label: 'Phone Number',
            type: 'tel',
            placeholder: 'e.g. +37312345678',
            pattern: '[+]373[6-7][0-9][0-9]{6}',
            required: true
          }
        ]
      }
    ],
    buttons: {
      backBtnText: '',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Select plan',
    title: 'Select your plan',
    subtitle: 'You have the option of monthly or yearly billing.',
    fieldGroups: [
      {
        type: 'plans',
        fields: [
          {
            iconType: 'arcade',
            name: 'Arcade',
            priceMonthly: 9,
            priceYearly: 90,
            currency: 'USD'
          },
          {
            iconType: 'advanced',
            name: 'Advanced',
            priceMonthly: 12,
            priceYearly: 120,
            currency: 'USD'
          },
          {
            iconType: 'pro',
            name: 'Pro',
            priceMonthly: 15,
            priceYearly: 150,
            currency: 'USD'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Add-ons',
    title: 'Pick add-ons',
    subtitle: 'Add-ons help enhance your gaming experience.',
    fieldGroups: [
      {
        type: 'addons',
        fields: [
          {
            title: 'Online service',
            description: 'Access to multiplayer games',
            priceMonthly: 1,
            priceYearly: 10,
            currency: 'USD'
          },
          {
            title: 'Larger storage',
            description: 'Extra 1TB of cloud save',
            priceMonthly: 2,
            priceYearly: 20,
            currency: 'USD'
          },
          {
            title: 'Customizable profile',
            description: 'Custom theme on your profile',
            priceMonthly: 2,
            priceYearly: 20,
            currency: 'USD'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Summary',
    title: 'Finishing up',
    subtitle: 'Double-check everything looks OK before confirming.',
    type: 'summary',
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Confirm'
    }
  },
  {
    name: 'Thank you!',
    title: 'Thank you!',
    subtitle: `Thanks for confirming your subscription! We hope you have fun using our platform.
      If you ever need support, please feel free to email us at support@loremgaming.com.`,
    type: 'thanks'
  }
];

export default formData;
