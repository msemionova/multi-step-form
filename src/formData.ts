const formData = [
  {
    name: 'Your info',
    title: 'Personal info',
    subtitle: 'Please provide your name, email address, and phone number.',
    fieldGroups: [
      {
        type: 'inputs',
        fields: [
          {
            type: 'text',
            label: 'Name',
            placeholder: 'John Doe'
          },
          {
            type: 'email',
            label: 'Email Address',
            placeholder: 'johndoe@email.com'
          },
          {
            type: 'tel',
            label: 'Phone Number',
            placeholder: 'e.g. +1 234 567 890'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: '',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Select plan',
    title: 'Select your plan',
    subtitle: 'You have the option of monthly or yearly billing.',
    fieldGroups: [
      {
        type: 'plans',
        fields: [
          {
            iconType: 'arcade',
            name: 'Arcade',
            priceMonthly: 9,
            priceYearly: 90,
            currency: 'USD'
          },
          {
            iconType: 'advanced',
            name: 'Advanced',
            priceMonthly: 12,
            priceYearly: 120,
            currency: 'USD'
          },
          {
            iconType: 'pro',
            name: 'Pro',
            priceMonthly: 15,
            priceYearly: 150,
            currency: 'USD'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Add-ons',
    title: 'Pick add-ons',
    subtitle: 'Add-ons help enhance your gaming experience.',
    fieldGroups: [
      {
        type: 'addons',
        fields: [
          {
            title: 'Online service',
            description: 'Access to multiplayer games',
            priceMonthly: 1,
            priceYearly: 10,
            currency: 'USD'
          },
          {
            title: 'Larger storage',
            description: 'Extra 1TB of cloud save',
            priceMonthly: 2,
            priceYearly: 20,
            currency: 'USD'
          },
          {
            title: 'Customizable profile',
            description: 'Custom theme on your profile',
            priceMonthly: 2,
            priceYearly: 20,
            currency: 'USD'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Summary',
    title: 'Finishing up',
    subtitle: 'Double-check everything looks OK before confirming.',
    type: 'summary',
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Confirm'
    }
  },
  {
    name: 'Thank you!',
    title: 'Thank you!',
    subtitle: `Thanks for confirming your subscription! We hope you have fun using our platform.
      If you ever need support, please feel free to email us at support@loremgaming.com.`,
    type: 'thanks'
  }
];

export default formData;
