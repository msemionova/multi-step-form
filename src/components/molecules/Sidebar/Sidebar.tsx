import { useContext } from 'react';
import activeStepContext from 'src/context/activeStepContext';
import styles from './Sidebar.module.scss';
import { SidebarProps } from './Sidebar.types';
import StepTitle from 'src/components/molecules/StepTitle/StepTitle';

export default function Sidebar({ data }: SidebarProps) {
  const { activeStep } = useContext(activeStepContext);

  return (
    <div className={styles.sidebar}>
      {
        data.map((step, index) =>
          <StepTitle
            key={Math.random()}
            number={index + 1}
            title={step.name as string}
            isActive={index === activeStep}
          />)
      }
    </div>
  );
}
