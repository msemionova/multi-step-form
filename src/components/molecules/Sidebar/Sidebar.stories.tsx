import type { Meta, StoryObj } from '@storybook/react';

import Sidebar from './Sidebar';
import formData from 'src/formData';

const meta: Meta<typeof Sidebar> = {
  component: Sidebar
};

export default meta;
type Story = StoryObj<typeof Sidebar>;

export const Default: Story = {
  args: {
    data: formData
  }
};
