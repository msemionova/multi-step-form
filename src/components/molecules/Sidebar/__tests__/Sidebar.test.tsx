import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import formData from 'src/formData';

import Sidebar from '../Sidebar';

describe('<Sidebar />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Sidebar data={formData}></Sidebar>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
