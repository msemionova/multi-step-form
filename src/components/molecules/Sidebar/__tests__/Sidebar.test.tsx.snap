// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly 1`] = `
<div
  className="sidebar"
>
  <div
    className="step"
  >
    <div
      className="circle active"
    >
      1
    </div>
    <div>
      <span>
        Step 1
      </span>
      <h2>
        Your info
      </h2>
    </div>
  </div>
  <div
    className="step"
  >
    <div
      className="circle"
    >
      2
    </div>
    <div>
      <span>
        Step 2
      </span>
      <h2>
        Select plan
      </h2>
    </div>
  </div>
  <div
    className="step"
  >
    <div
      className="circle"
    >
      3
    </div>
    <div>
      <span>
        Step 3
      </span>
      <h2>
        Add-ons
      </h2>
    </div>
  </div>
  <div
    className="step"
  >
    <div
      className="circle"
    >
      4
    </div>
    <div>
      <span>
        Step 4
      </span>
      <h2>
        Summary
      </h2>
    </div>
  </div>
  <div
    className="step"
  >
    <div
      className="circle"
    >
      5
    </div>
    <div>
      <span>
        Step 5
      </span>
      <h2>
        Thank you!
      </h2>
    </div>
  </div>
</div>
`;
